Description:  Check that $tmpfolder/$station exists.
 .
 Otherwise, grep returns a "No such file or directory" error if it doesn't.
Author: Jari Aalto <jari.aalto@cante.net>
Author: Doug Torrance <dtorrance@monmouthcollege.edu>
Last-Update: 2014-08-24

--- a/Src/weather.pl
+++ b/Src/weather.pl
@@ -367,32 +367,35 @@
 
 # Check for ok temperature
 
-open GREP, "grep Temp $tmpfolder/${station} | ";
-$templine = <GREP>;
+if ( -f "$tmpfolder/$station" )
+{
+    open GREP, "grep Temp $tmpfolder/${station} | ";
+    $templine = <GREP>;
 
-$templine = ~m/^Temp:(.*)$/;
+    $templine = ~m/^Temp:(.*)$/;
 
-{
-    if ( $temp > -50  and  $temp < 150 )
     {
-	if ( $lasttemp > -50  and  $lasttemp < 150 )
+	if ( $temp > -50  and  $temp < 150 )
 	{
-	    if ( $temp > -2  and  $temp < 2 )
+	    if ( $lasttemp > -50  and  $lasttemp < 150 )
 	    {
-		if ( $lasttemp > 7  or  $lasttemp < -7 )
+		if ( $temp > -2  and  $temp < 2 )
 		{
-		    $station = "";	# probably invalid
+		    if ( $lasttemp > 7  or  $lasttemp < -7 )
+		    {
+			$station = "";	# probably invalid
+		    }
 		}
 	    }
 	}
+	else
+	{
+	    $station = "";
+	}
     }
-    else
-    {
-	$station = "";
-    }
-}
 
-close GREP;
+    close GREP;
+}
 
 debug("Station: $station");
 
